### PR TITLE
Feature/bme280

### PR DESCRIPTION
**Pull request by strongly-typed**

### CODE REVIEW

File: src/xpcc/driver/pressure/bme280.hpp [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+/**
+ * BME280 digital absolute pressure sensor driver
+ *
+ * The BME280 is a high precision digital temperature, pressure and humidity sensor with I2C interface.
+ *
+ * It is known that many BME280-modules suffer from self-heating (I2C pull-ups?). 
+ * So the reported temperature is about 1 to 3 degree higher than with other sensors.
+ * Forced ventilation can drop the temperature by 2 to 3 degrees.
+ *
+ * @author	Niklas Hauser
```
Comment (salkinium) [inline src/xpcc/driver/pressure/bme280.hpp]: Nope.

File: src/xpcc/driver/pressure/bme280.hpp [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+		FILTER_1 = Bit3,
+		FILTER_0 = Bit2,
+		SPI3W_EN = Bit0,
+	};
+	XPCC_FLAGS8(Config);
+
+	enum class
+	TimeStandby : uint8_t
+	{
+		Mask  = int(Config::T_SB_2) | int(Config::T_SB_1) | int(Config::T_SB_0),
```
Comment (salkinium) [inline src/xpcc/driver/pressure/bme280.hpp]: [`Configuration` can give you a mask](http://xpcc.io/api/structxpcc_1_1_configuration.html#a9468f21043bc25306f034af461687204), this is not necessary, unless you explicityly use it somewhere.

File: src/xpcc/driver/pressure/bme280_impl.hpp [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	// 67 01 00 14 06 00 1E (hexadecimal) shall be converted to
+	// H2=359, H3=0, H4=326, H5=0, H6=30 (decimal)
+
+	buffer[0] = i(Register::CALIB26);
+
+	this->transaction.configureWriteRead(buffer, 1, calBuffer, 7);
+
+	if (RF_CALL( this->runTransaction() ))
+	{
+		data.calibration.H2 = xpcc::fromLittleEndian( int16_t( (int16_t(calBuffer[1] << 8)) | calBuffer[0]));
```
Comment (salkinium) [inline src/xpcc/driver/pressure/bme280_impl.hpp]: I think `(calBuffer[1] << 8) | calBuffer[0]` already does an endianness conversion.

File: src/xpcc/driver/pressure/bme280_impl.hpp [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+		data.calibration.H3 = calBuffer[2];
+		data.calibration.H4 = xpcc::fromLittleEndian( int16_t( (int16_t(calBuffer[3] << 4)) | (int16_t(calBuffer[4] & 0x0f)) ) );
```
Comment (salkinium) [inline src/xpcc/driver/pressure/bme280_impl.hpp]: Probably one `int16_t` too many.

**Review by salkinium — COMMENTED** [review]
Bosch makes complcated sensors…

**Comment by salkinium** [comment]
CI failed because I fail at setting correct DNS records.

File: src/xpcc/driver/pressure/bme280_impl.hpp (hunk 4 above)
Comment (strongly-typed) [reply src/xpcc/driver/pressure/bme280_impl.hpp]: Simplified and tested.

File: src/xpcc/driver/pressure/bme280.hpp (hunk 1 above)
Comment (strongly-typed) [reply src/xpcc/driver/pressure/bme280.hpp]: It's heavily based on your BMP085 driver so it's just a copy-paste ;-)

**Review by salkinium — APPROVED** [review]

**Comment by salkinium** [comment]
TIL: don't search for "blue pill" on ebay.

**Pull request merged**